File: HackerRank/Algorithms/Warmup/Simple_Array_Sum.cs
using System.IO;
using System.Linq;
using NUnit.Framework;

public class Simple_Array_Sum 
{
    private TextReader _in;
    private TextWriter _out;

    public Simple_Array_Sum(TextReader IN, TextWriter OUT)
    {
        _out = OUT;
        _in = IN;
    }

    public void Resolve()
    {
        var nb = int.Parse(_in.ReadLine());

        _out.WriteLine(_in.ReadLine()
            .Split(' ')
            .Select(int.Parse)
            .Sum()
            .ToString());
    }
}

[TestFixture]
public class Test
{
    [Test]
    public void GetSimpleSumArray()
    {
        var input = @"6
1 2 3 4 10 11";

        var output = new StringWriter();

        new Simple_Array_Sum(new StringReader(input), output).Resolve();

        Assert.AreEqual("31\r\n",output.GetStringBuilder().ToString());
    }
}

File: HackerRank_FS/Simple_Array_Sum.fs
namespace HackerRank_FS.Algorithms.Warmup
//Sample Input
//6
//1 2 3 4 10 11

//Sample Output
//31

open NUnit.Framework
open FsUnit
open System
open System.IO

type SimpleArraySum(IN:TextReader, OUT:TextWriter) = 
    member this.Sum = 
        let nb = IN.ReadLine() |> int
        OUT.Write (IN.ReadLine().Split [|' '|] |> Seq.map int |> Seq.sum)


[<TestFixture>]
type `` Given a simple array sum ``() =
    let out = new StringWriter()

    [<Test>]
    member x.``When try sample use case ``() =
        let simpleArraySum = new SimpleArraySum(new StringReader(@"6
1 2 3 4 10 11"),out)
        simpleArraySum.Sum
        Assert.AreEqual("31",out.GetStringBuilder().ToString())
